Automattic/wp-calypso
# Post Revisions: Use diffs from the API instead of diffing locally on the front end

The current implementation computes the diff between revisions in the browser / device.

This uses the diffs computed by the REST API server instead. As a result, it is enabled on mobile devices now and performs a lot better than the client-side diffing.

fixes #20583 as well

## To Test:

### Manual testing

* ~~Apply API patch in D8615~~ endpoint is live!
* Run this branch
* Edit a post with revisions
* Click the History button
  * Diffs should be shown and UI fully functional
  * Clicking `Load` should put the "compare from" revision's title & content into the Editor
* Repeat with the `?flags=-post-editor/revisions` query arg
  * The link to wp-admin revisions should be visible 
* Test on Mobile
  * It will show the `History` button in landscape orientation & function well.
  * Rotating back to portrait mode will still be functional, but with the list of revisions scrolling horizontally at the top of the modal

### Automated testing

`npm run test-client:watch client/state/posts/revisions client/state/selectors/test/*revision*`
